### DISCUSSION TITLE
BigInt field type possible?

### DISCUSSION THREAD

**Discussion by JSatterly** [oqtane/oqtane.framework#2806] [General]
In the .Server.Migrations.EntityBuilders.BaseEntityBuilder, there is an ability to add field types for boolean, integer, string, maxstring. Is there an ability to build a tiny integer or BigInt field type? Or is it just suggested to use Int with these?

**Comment by leigh-pointer** [comment]
@JSatterly BigInt or Tiny is not support out of the box but you could use Value Conversions https://learn.microsoft.com/en-us/ef/core/modeling/value-conversions?tabs=data-annotations

**Reply by JSatterly** [reply]
Thanks for the link, these will come in handy.

**Comment by sbwalker** [comment]
@JSatterly In general, the rule when developing a framework which supports multiple types of databases is to avoid any database-specific features. This is the reason why Oqtane only provides support for the most common data types, as support for more obscure data types varies greatly from one database engine to another. There is a database provider model so that custom behavior can be added for each specific database but it can get complicated. For example, if it was necessary to add support for BigInt we would need to research if SQL Server, MySQL, PostgreSQL, and SQLite all support that data type... and in the cases where they do not, a fallback strategy would need to be included. The problem with fallback strategies is that they still have different run-time behaviors which can cause issues in the future.

**Comment by JSatterly** [comment]
Ahh gotcha, I'll go for generic then, thanks.
